config/migrate.go

```diff
@@ -0,0 +1,200 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+//go:generate $GOROOT/bin/go run ./defaults_gen/defaults_gen.go -h ../scripts/LICENSE_HEADER -p config -o ./local_defaults.go -j ../installer/config.json.example
+//go:generate $GOROOT/bin/go fmt local_defaults.go
+
+// AutogenLocal - this variable is the "input" for the config default generator which automatically updates the above defaultLocal varaible.
+// it's implemented in ./config/defaults_gen.go, and should be the only "consumer" of this exported variable
+var AutogenLocal = getVersionedDefaultLocalConfig(getLatestConfigVersion())
```

Comment:
Why is it done this way? the generator might call `getVersionedDefaultLocalConfig(getLatestConfigVersion())` or `getVersionedDefaultLocalConfig(version)` with version from the command line.

Reply:
The generator is in a separate package, and cannot access these methods directly. Sure, we could expose the functions - but then we're other parties as well. Maybe this should be refactored using internal package ? I would rather finish this pass before refactoring it further, as the gained value diminish.